website_mass_mailing_name/static/src/js/website_mass_mailing_name.js

```diff
@@ -27,7 +28,9 @@ odoo.define("website_mass_mailing_name.subscribe", function (require) {
                 };
             // Stop on error
             if (email_error || name_error) {
-                this.$target.addClass("has-error")
+                this.$target.addClass("o_has-error");
```

Comment:
The old `has-error` class came from Bootstrap 3 and you shouldn't be using it anymore. It's probably still working, but only [due to a backwards-compatibility layer that Odoo 12.0 has](https://github.com/odoo/odoo/blob/c52b9dfc3761c166f5f1b475e7a14ab2ec9fb66f/addons/website/static/src/scss/compatibility/bs3_for_12_0.scss#L81-L83).

The class `o_has-error` you are using... I cannot find it in the source code... Where does it come from?

In any case, please check [this section about form validation in BS4](https://getbootstrap.com/docs/4.3/components/forms/#how-it-works). I think you can use the constraint validation API, or the classes `is-valid` and `is-invalid` (although this is more a fallback for server-side validation; not your case).

Reply:
I found it by inspect elements but true is no way it comes from Odoo. So I will change it as you suggest and test again